discordbot/cogs/channels.py

```diff
@@ -0,0 +1,196 @@
+            else:
+                await ctx.send(f"Channel '{name}' does not exist.")
+                
+    
+    @commands.command(brief='Create or delete a team category and associated channels', aliases=[])
+    @commands.has_permissions(administrator=True)
+    async def team(self, ctx, action, *, name=None):
+        guild = ctx.guild
+
+        if action.lower() not in ['create', 'delete','del']:
+            await ctx.send('Invalid action. Please try again: `!team create [team name]` or `!team delete [team name]`')
+            return
+        if name is None:
+            await ctx.send('You did not include a name for your team. Please try again: `!team create [team name]` or `!team delete [team name]`')
+            return
+
+        if action.lower() == "create":
+            role = discord.utils.get(ctx.guild.roles, name=name)
+            if not role:
+                await ctx.send(f"The role {name} does not exist.")
+                return
+
+            category_name = name + ' team'
+            category = discord.utils.get(guild.categories, name=category_name)
+            if category:
+                await ctx.send(f"A category with the name '{category_name}' already exists.")
+                return
+
+            await ctx.send("Would you like to customize the voice and text channel names? y/n? (Default names would be applied otherwise)")
+            default = True
+            try:
+                customcheck = await self.bot.wait_for('message', check=lambda message: message.author == ctx.author, timeout=60)
+                if customcheck.content.upper() == 'Y':
+                    default = False
+            except TimeoutError:
+                await ctx.send("Timed out. Please try the command again.")
+                return
+
+            if default:
+                text_channel_name = name.replace(' ', '-') + '-general'
+                voice_channel_name = name.replace(' ', '-') + '-meeting'
+            else:
+                await ctx.send("Please enter the name for the text channel:")
+                try:
+                    text_channel_name = await self.bot.wait_for('message', check=lambda message: message.author == ctx.author, timeout=60)
+                    text_channel_name = text_channel_name.content
+                except TimeoutError:
+                    await ctx.send("Timed out. Please try the command again.")
+                    return
+
+                await ctx.send("Please enter the name for the voice channel:")
+                try:
+                    voice_channel_name = await self.bot.wait_for('message', check=lambda message: message.author == ctx.author, timeout=60)
+                    voice_channel_name = voice_channel_name.content
+                except TimeoutError:
+                    await ctx.send("Timed out. Please try the command again.")
+                    return
+
+            category = await guild.create_category_channel(name=category_name)
```

Comment:
I note that the category itself never has its permissions changed, correct me if I am wrong, but I would assume this means that everyone would see an empty category.

Reply:
Also, very sorry for how late this review came, I had assumed that Aarush was doing it.

Reply:
await category.set_permissions needs to be called with overwrites I think